War Within patches and updates for review

Thrumbar opened the pull request.

atk commented on Carbonite/NxMap.lua:
```diff
@@ -227,7 +230,8 @@ function SetMapToCurrentZone(carbcalled)
 end
 
-function SetMapByID(zone,level)
+--DeaTHCorE - rename of the func, so we can call this save
```
I believe the original function is deliberately overwritten, so it will grab the calls of other addons. Bugfixes shouldn't change intended behaviour, as far as I'm concerned.

deathcore01 commented:
i think not, that the original func for grab the calls of other addons overwritten is, it was overwritten for the variable Nx.Map.RMapId. and a overwrite, that causes in incompatibility or errors is not useful ;)

Rythal commented:
No, it was overwritten for exactly ask atk said. With addons like dugi or zygor also making mass amounts of setmapbyid calls along with Carbonite doing the same it was causing people to get kicked from the game (literally) so I introduced my own throttling mechanisms to make every and all addons play nice not just carbonite.

deathcore01 commented:
hmm okay, so can close this and i will use this for my local copy only, i can say, its work without any problem and i use dugi by my char's, who i level - by the chars, that have level 100 reached, i not need dugi and i have deactivated it, with dugi is the only situation, who i have this issue :(

Rythal commented:
If it's causing a problem, it's more finding out why... the base idea behind it is a simple check

if I receive a call to change the map, check what map is being displayed, if it's the same one i'm already on do nothing... if it's a new map then change it.

There's nothing there that should of broken Dugi, and if it is then it needs looked into of just why it happened.

deathcore01 commented:
hmm good idea, i will check this tomorrow...i check the numbers of the calls from setmapbyid with and without dugi and then i will see.

Thrumbar committed:
War Within Update 1

Thrumbar committed:
War Within Update 2

Thrumbar committed:
War Within Update 3

Thrumbar committed:
War Within fix for update 3

Thrumbar committed:
War Within update 4

IrcDirk commented:
Ill wait till pre-patch to merge this.

Thrumbar committed:
patch for below issue related to below error

Getting this error when tracking quest from quest tracker

318x bad argument #1 to '?' (Usage: local spellInfo = C_Spell.GetSpellInfo(spellIdentifier))
[string "=[C]"]: in function `GetSpellInfo'
[string "@Carbonite/NxTravel.lua"]:421: in function `MakePath'
[string "@Carbonite/NxMap.lua"]:5941: in function `CalcTracking'
[string "@Carbonite/NxMap.lua"]:9690: in function `GotoPlayer'
[string "@Carbonite.Quests/NxQuest.lua"]:10547: in function `TrackOnMap'
[string "@Carbonite.Quests/NxQuest.lua"]:10083: in function `Set'
[string "@Carbonite.Quests/NxQuest.lua"]:9976: in function `UserFunc'
[string "@Carbonite/NxUI.lua"]:6538: in function `UserFunc'
[string "@Carbonite/NxUI.lua"]:3705: in function <Carbonite/NxUI.lua:3687>

Locals:
(*temporary) = "bad argument #1 to '?' (Usage: local spellInfo = C_Spell.GetSpellInfo(spellIdentifier))"

Thrumbar commented:
War Within Update 2 has an issue in that the use item button on the quest watch window is not working,  

Errors is was intended to fix,,
1x Carbonite/Carbonite.xml:136 Frame NxListFrms1: Unknown function QuestObjectiveItem_OnLoad in element OnLoad
1x Carbonite/Carbonite.xml:137 Frame NxListFrms1: Unknown function QuestObjectiveItem_OnEvent in element OnEvent
1x Carbonite/Carbonite.xml:139 Frame NxListFrms1: Unknown function QuestObjectiveItem_OnShow in element OnShow
1x Carbonite/Carbonite.xml:140 Frame NxListFrms1: Unknown function QuestObjectiveItem_OnHide in element OnHide
1x Carbonite/Carbonite.xml:136 Frame NxListFrms2: Unknown function QuestObjectiveItem_OnLoad in element OnLoad
1x Carbonite/Carbonite.xml:137 Frame NxListFrms2: Unknown function QuestObjectiveItem_OnEvent in element OnEvent
1x Carbonite/Carbonite.xml:139 Frame NxListFrms2: Unknown function QuestObjectiveItem_OnShow in element OnShow
1x Carbonite/Carbonite.xml:140 Frame NxListFrms2: Unknown function QuestObjectiveItem_OnHide in element OnHide

Thrumbar committed:
Updates

Fix for quest log item use and some updates and reversions

Thrumbar committed:
Fix for accidently removed end

Thrumbar commented:
Can these be pushed into the ptr branch
